HeartbeatLogging/Sources/HeartbeatStorage.swift

```diff
@@ -14,57 +14,86 @@
 
 import Foundation
 
-/// A type that can store and remove hearbeats.
 protocol HeartbeatStorageProtocol {
-  func offer(_ heartbeat: Heartbeat)
-  func flush() -> HeartbeatInfo?
+  typealias HeartbeatInfoTransform = (HeartbeatInfo?) -> HeartbeatInfo?
+
+  func async(_ transform: @escaping HeartbeatInfoTransform)
```

Comment:
Could be confusing to use a Swift keyword as a function name.  Is there something more descriptive?

Reply:
I like the more verbose `readAndWriteAsync` version so far as it saves the reader time since more information can be get from code without looking at API docs.

Reply:
Thanks @maksymmalyhin. I think I'll go with:
```swift
func readAndWriteAsync(using transform: @escaping HeartbeatInfoTransform)
```